lib/rubocop/cop/rspec/repeated_subject_call.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Checks for repeated calls to subject missing that it is memoized.
+      #
+      # @example
+      #   # bad
+      #   it do
+      #     subject
+      #     expect { subject }.to not_change { A.count }
+      #   end
+      #
+      #   it do
+      #     expect { subject }.to change { A.count }
+      #     expect { subject }.to not_change { A.count }
+      #   end
+      #
+      #   # good
+      #   it do
+      #     expect { my_method }.to change { A.count }
+      #     expect { my_method }.to not_change { A.count }
+      #   end
+      #
+      class RepeatedSubjectCall < Base
+        BLOCK_MSG = 'Calls to subject are memoized, this block is misleading'
+
+        # @!method subject_calls(node)
+        def_node_search :subject_calls, <<~PATTERN
+          (send nil? :subject)
```

Comment:
I think we need to detect named subjects and memoized helpers in general, too.

Reply:
Agreed - wasn't sure on the best way to achieve that especially given the complex searching we already do below with just `subject`.

Reply:
[This](https://github.com/rubocop/rubocop-rspec/blob/e609b2a655ed9609ec8cc9287a5c9ad877c6aaf4/lib/rubocop/cop/rspec/subject_stub.rb#L71) should work.